src/pubsub/ua_pubsub.c

```diff
@@ -993,46 +993,55 @@ UA_DataSetWriter_generateDataSetMessage(UA_Server *server, UA_DataSetMessage *da
     return UA_STATUSCODE_GOOD;
 }
 
-/*
- * This callback triggers the collection and publish of NetworkMessages and the contained DataSetMessages.
- */
+/* This callback triggers the collection and publish of NetworkMessages and the
+ * contained DataSetMessages. */
 void
 UA_WriterGroup_publishCallback(UA_Server *server, UA_WriterGroup *writerGroup) {
     if(!writerGroup){
-        UA_LOG_ERROR(server->config.logger, UA_LOGCATEGORY_SERVER, "Publish failed. WriterGroup not found");
+        UA_LOG_ERROR(server->config.logger, UA_LOGCATEGORY_SERVER,
+                     "Publish failed. WriterGroup not found");
         return;
     }
+
+    /* Nothing to do? */
```

Comment:
If all WGs from a PubSub Connection are removed there can't be produced any DataSetMessages. Maybe we should later detect this case in the remove writegroup and remove the publish callback.